terraform/modules/irobot/main.tf

```diff
@@ -0,0 +1,97 @@
+variable "flavour" {}
+variable "domain" {}
+variable "network_id" {}
+variable "count" {}
+
+variable "security_group_ids" {
+  type    = "map"
+  default = {}
+}
+
+variable "key_pair_ids" {
+  type    = "map"
+  default = {}
+}
+
+variable "image" {
+  type    = "map"
+  default = {}
+}
+
+variable "bastion" {
+  type    = "map"
+  default = {}
+}
+
+resource "openstack_networking_floatingip_v2" "irobot" {
+  provider = "openstack"
+  pool     = "nova"
+  count    = "${var.count}"
+}
+
+resource "openstack_compute_instance_v2" "irobot" {
+  provider    = "openstack"
+  count       = "${var.count}"
+  name        = "irobot-${count.index + 1}"
+  image_name  = "${var.image["name"]}"
+  flavor_name = "${var.flavour}"
+  key_pair    = "${var.key_pair_ids["mercury"]}"
+
+  security_groups = [
+    "${var.security_group_ids["ping"]}",
+    "${var.security_group_ids["ssh"]}",
+    "${var.security_group_ids["http"]}",
+    "${var.security_group_ids["iRods"]}",
+  ]
```

Comment:
The setup will require access for Kerberos but that can be added when you get there.

Reply:
yes, I thought about that. setting up Kerberos auth is a different task, not sure if I'm meant to do it